File: packages/@simpli/cli-scaffold/generator/template/src/model/response/ItemDoPrincipalResp.ts
/**
 * ItemDoPrincipalResp
 * @author martinlabs CRUD generator
 */
import {ID, TAG, Resource} from '@/simpli'
import {ResponseSerialize} from '@/simpli'

import ItemDoPrincipal from '@/model/ItemDoPrincipal'
import Principal from '@/model/Principal'

export default class ItemDoPrincipalResp extends Resource {
  readonly $endpoint: string = '/Crud/ItemDoPrincipal{/id}'

  @ResponseSerialize(ItemDoPrincipal)
  itemDoPrincipal: ItemDoPrincipal = new ItemDoPrincipal()

  @ResponseSerialize(Principal)
  allPrincipal: Principal[] = []

  get $id() {
    return this.itemDoPrincipal.$id
  }

  set $id(val: ID) {
    this.itemDoPrincipal.$id = val
  }

  get $tag() {
    return this.itemDoPrincipal.$tag
  }

  set $tag(val: TAG) {
    this.itemDoPrincipal.$tag = val
  }
}

File: packages/@simpli/cli-server/generator/template/src/main/java/br/com/martinlabs/usecase/crud/response/ItemDoPrincipalResp.kt
package br.com.martinlabs.usecase.crud.response

import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty
import br.com.martinlabs.usecase.model.ItemDoPrincipal
import br.com.martinlabs.usecase.model.Principal

@ApiModel(value = "ItemDoPrincipalResp")
class ItemDoPrincipalResp {

    var itemDoPrincipal: ItemDoPrincipal? = null

    @ApiModelProperty(value = "Possible Principal values")
    var allPrincipal: MutableList<Principal>? = null
}
